test/testenv/rancher.go

```diff
@@ -414,3 +416,62 @@ func deployNgrokIngress(ctx context.Context, input RancherDeployIngressInput) {
 	By("Setting up default ingress class")
 	Expect(input.BootstrapClusterProxy.Apply(ctx, input.DefaultIngressClassPatch, "--server-side")).To(Succeed())
 }
+
+type PreRancherInstallHookInput struct {
+	Ctx                context.Context
+	RancherInput       *DeployRancherInput
+	PreSetupOutput     PreManagementClusterSetupResult
+	SetupClusterResult *SetupTestClusterResult
+	E2EConfig          *clusterctl.E2EConfig
+}
+
+type PreRancherInstallHookResult struct {
+	HostName string
+}
+
+// PreRancherInstallHook is a hook that can be used to perform actions before Rancher is installed.
+func PreRancherInstallHook(input *PreRancherInstallHookInput) PreRancherInstallHookResult {
```

Comment:
I don't like how part of the structure is defined within the suite, and part in a general purpose hook. This might lead to maintainability issues.